nifi-nar-bundles/nifi-media-bundle/nifi-media-processors/src/main/java/org/apache/nifi/processors/media/ExtractMediaMetadata.java

```diff
@@ -0,0 +1,311 @@
+                    + " added to the flowfile attributes.  If left blank, all metadata keys parsed will be added to the"
+                    + " flowfile attributes.")
+            .required(false)
+            .addValidator(StandardValidators.REGULAR_EXPRESSION_VALIDATOR)
+            .build();
+
+    static final PropertyDescriptor METADATA_KEY_PREFIX = new PropertyDescriptor.Builder()
+            .name("Metadata Key Prefix")
+            .description("Text to be prefixed to metadata keys as the are added to the flowfile attributes.  It is"
+                    + " recommended to end with with a separator character like '.' or '-', this is not automatically "
+                    + " added by the processor.")
+            .required(false)
+            .addValidator(StandardValidators.ATTRIBUTE_KEY_VALIDATOR)
+            .expressionLanguageSupported(true)
+            .build();
+
+    static final PropertyDescriptor CONTENT_BUFFER_SIZE = new PropertyDescriptor.Builder()
+            .name("Content Buffer Size")
+            .description("The size for media content buffer during processing, or -1 for unlimited.  If not"
+                    + " provided, the underlying parser default is used.")
+            .required(false)
+            .addValidator(StandardValidators.INTEGER_VALIDATOR)
+            .addValidator(new Validator() {
+                @Override
+                public ValidationResult validate(String subject, String input, ValidationContext context) {
+                    Integer val = null;
+                    try {
+                        val = Integer.parseInt(input);
+                    } catch (NumberFormatException ignore) {
+                    }
+                    return new ValidationResult.Builder()
+                            .subject(subject)
+                            .input(input)
+                            .valid(val != null && val >= -1)
+                            .explanation(subject + " must be a valid integer equal to or greater than -1.")
+                            .build();
+                }
+            })
+            .expressionLanguageSupported(true)
+            .build();
+
+    static final Relationship SUCCESS = new Relationship.Builder()
+            .name("success")
+            .description("Any FlowFile that successfully has media metadata extracted will be routed to success")
+            .build();
+
+    static final Relationship FAILURE = new Relationship.Builder()
+            .name("failure")
+            .description("Any FlowFile that fails to have media metadata extracted will be routed to failure")
+            .build();
+
+    private Set<Relationship> relationships;
+    private List<PropertyDescriptor> properties;
+
+    private final AtomicReference<Pattern> mimeTypeFilterRef = new AtomicReference<>();
+    private final AtomicReference<Pattern> metadataKeyFilterRef = new AtomicReference<>();
+
+    private volatile AutoDetectParser autoDetectParser;
```

Comment:
It's not needed until it's scheduled, and this allows a processor stop and start to get a new parser if there are problems.